admin api

indeyets opened the pull request:
### Added
- Administrative API. This API (with `/api/admin` prefix) is not a part of official and supported/versioned server API.

  Some users can have administrative roles: *administrator* and/or *moderator*. The *administrator* role can be assigned only by site operator with manual access to database. *Administrators* can appoint *moderators*. *Moderators* can perform some site management actions, for now it is the freeze-related actions.

  The following methods have been added:
  - Allowed for any administrative user:
    - `GET /api/admin/whoami` — info about user himself, in particular, the roles he has.
    - `GET /api/admin/journal` — journal of all administrative actions.
  - Allowed for *administrators* only:
    - `GET /api/admin/members` — list of all administrative users.
    - `POST /api/admin/members/:username/promote` — gives user a *moderator* role.
    - `POST /api/admin/members/:username/demote` — removes *moderator* role from user.
  - Allowed for *moderators* only:
    - `GET /api/admin/users/frozen` — list of users who are currently frozen.
    - `POST /api/admin/users/:username/freeze` — freeze user until a certain time.
    - `POST /api/admin/users/:username/unfreeze` — unfreeze user.

indeyets committed:
Add migration for creating admin-related storage

indeyets committed:
Methods to check if User is moderator or admin

indeyets committed:
Separate root-router from vN-router

indeyets committed:
Register namespaces for admin/moderator urls

indeyets committed:
Use proper CORS implementation

davidmz committed:
Create /api/admin/whoami and /api/admin/members/* methods

davidmz committed:
Disable triggers during DB cleanup

davidmz committed:
Add /api/admin/journal endpoint and create journal entries

davidmz committed:
Update changelog

davidmz committed:
Filter /api/admin/ from routes coverage test

davidmz commented:
Well, there is an admin API. And I cannot suggest @indeyets for review because he is a PR owner:)

davidmz committed:
Require 'details' in 'admin_actions' to be an object

davidmz committed:
Add admin methods for users freezing

davidmz committed:
Add 'doInTransaction' dbAdapter helper

This helper runs the specified function in a transaction, substituting dbAdapter.database with the current transaction. This allows you to use most dbAdapter methods inside the function.

davidmz committed:
Run administrative actions and write to journal in transactions
